rsconnect/bundle.py

```diff
@@ -507,6 +509,92 @@ def make_api_manifest(
     return manifest, relevant_files
 
 
+def make_html_fileslist(
+    path,  # type: str
+    entrypoint,  # type: str
+    extra_files=None,  # type: typing.Optional[typing.List[str]]
+    excludes=None,  # type: typing.Optional[typing.List[str]]
+):
+    # type: (...) -> typing.Tuple[typing.Dict[str, typing.Any], typing.List[str]]
+    """
+    Makes a manifest for static html deployment.
+
+    :param path: the file, or the directory containing the files to deploy.
+    :param entry_point: the main entry point for the API.
+    :param extra_files: a sequence of any extra files to include in the bundle.
+    :param excludes: a sequence of glob patterns that will exclude matched files.
+    :return: the manifest and a list of the files involved.
+    """
+    entrypoint = entrypoint or infer_entrypoint(path, "text/html")
+    if is_environment_dir(path):
+        excludes = list(excludes or []) + ["bin/", "lib/"]
+
+    relevant_files = _create_api_file_list(path, "", extra_files, excludes)
+    manifest = make_html_manifest(entrypoint)
+
+    for rel_path in relevant_files:
+        manifest_add_file(manifest, rel_path, path)
+
+    return manifest, relevant_files
+
+
+def infer_entrypoint(path, mimetype):
+    if os.path.isfile(path):
+        return path
+    if not os.path.isdir(path):
+        raise ValueError("Entrypoint is not a valid file type or directory.")
+
+    default_mimetype_entrypoints = {"text/html": "index.html"}
+    if mimetype not in default_mimetype_entrypoints:
+        raise ValueError("Not supported mimetype inference.")
+
+    entrypoint_candidates = []
+    mimetype_dict = defaultdict(list)
+
+    for subdir, dirs, files in os.walk(path):
+        for file in files:
+            abs_path = os.path.join(subdir, file)
+            rel_path = os.path.relpath(abs_path, path)
+            mimetype_dict[guess_type(file)[0]].append((file, rel_path))
+
+    for file, rel_path in mimetype_dict[mimetype]:
+        if file in default_mimetype_entrypoints[mimetype]:
+            entrypoint_candidates.append(rel_path)
+
+    return entrypoint_candidates.pop() if len(entrypoint_candidates) == 1 else None
```

Comment:
Otherwise, if I understand correctly, `mimetype_dict[mimetype]` is a list of the HTML files. If that list contains 1 item then it is the default entrypoint, otherwise we cannot choose automatically.